kernel/sched.c

```diff
@@ -658,6 +658,22 @@ void _update_time_slice_before_swap(void)
 }
 #endif /* CONFIG_TIMESLICING */
 
+void _ready_thread(struct k_thread *thread)
+{
+	if (_is_thread_ready(thread)) {
+		_add_thread_to_ready_q(thread);
+	}
+
+#if defined(CONFIG_TICKLESS_KERNEL) && !defined(CONFIG_SMP)
+	if (_is_thread_time_slicing(_get_next_ready_thread())) {
+		_set_time(_time_slice_duration);
+		return;
+	}
+#endif
+
+	sys_trace_thread_ready(thread);
```

Comment:
move the trace hook up and before returning.

Reply:
Whoops, yeah, that was a cut & paste bug.  There's no utility in the early return, and I guess tracing hooks are invisible to my eyes.  Fixed.

Reply:
does not look it was fixed